scripts/install.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+
+ROOT_DIR="$(pwd)"
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    OS="macOS"
+    NUMBER_OF_PROCESSORS="$(sysctl -n hw.ncpu)"
+fi
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    OS="Linux"
+    NUMBER_OF_PROCESSORS="$(nproc)"
+fi
+
+GENERATOR="Unix Makefiles"
+
+SUBMODULE_DIR="${ROOT_DIR}/vendor"
+SUBMODULE_INCLUDE_DIR="${SUBMODULE_DIR}/include"
+SUBMODULE_LIB_DIR="${ROOT_DIR}/lib/${OS}"
```

Comment:
might be a useful addition to the file structure

Reply:
I'm a little unsure what this means